### ISSUE TITLE
Enforce variable usage for certain declarations matching a pattern + enforce custom media usage

### ISSUE THREAD

**Issue by cphoover** [stylelint/stylelint#3560]
> What is the problem you're trying to solve?

Would be nice if there were built in rules for forcing that a variable (using either `var(--syntax)` or `$less-sass-variable-syntax`) for declarations.

For example I never want ui devs on my team to write hex colors directly to color declarations.

I see this 3p rule https://github.com/sh-waqar/stylelint-declaration-use-variable but it would be nice if this were a built in feature.

Also I would like to see a rule to enforce the usage of custom media query aliases.

In other words I don't want component styles to every define their own media queries, they should be forced to use global definitions with the `@custom-media` syntax.

**Comment by jeddy3** [comment]
@cphoover Thanks for the requests and for using the template.

> I see this 3p rule https://github.com/sh-waqar/stylelint-declaration-use-variable but it would be nice if this were a built in feature.

It's a plugin because it targets non-standard constructs like dollar variables, whereas the built-in rules focus on standard constructs only. It doesn't meet our [criteria for inclusion](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#criteria-for-inclusion). The plugin system is easy to use and specifically built so that stylelint is extensible. It should be embraced.

You might also get some mileage using the built-in whitelist/blacklist rules e.g.

```
"declaration-property-value-whitelist": {
  color: ["/^var(--"]
 }
```

But use the plugin or write your own if you want something more customised.

> Also I would like to see a rule to enforce the usage of custom media query aliases.

You can use `"media-feature-name-whitelist": [],` to enforce this. This will disallow the use of `min-width` and `max-width` media features, but allow custom media query _extension_ names to be used as they are [used in a boolean context](https://drafts.csswg.org/mediaqueries-5/#custom-mq) and not checked my this rule.
